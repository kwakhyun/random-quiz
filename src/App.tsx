import { QuizProvider } from './context/QuizContext';
import { Global, ThemeProvider } from '@emotion/react';
import theme from './styles/theme';
import { global } from './styles';
import Router from './router';

const App = () => {
  return (
    <QuizProvider>
      <ThemeProvider theme={theme}>
        <Global styles={global} />
        <Router />
      </ThemeProvider>
    </QuizProvider>
  );
};

export default App;
