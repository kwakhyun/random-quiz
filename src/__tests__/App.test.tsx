import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import App from '../App';

test('시작 페이지 테스트', () => {
  const appElement = render(<App />);
  expect(appElement).toMatchSnapshot();

  const titleElement = screen.getByText(/Random Quiz!/i);
  expect(titleElement).toBeInTheDocument();

  const contentElement = screen.getByText(/아래 버튼을 누르면 퀴즈 풀기를 시작합니다./i);
  expect(contentElement).toBeInTheDocument();
});
