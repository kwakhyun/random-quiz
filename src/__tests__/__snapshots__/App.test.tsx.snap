// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`시작 페이지 테스트 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1itlo10"
      >
        <h1>
          Random Quiz!
        </h1>
        <p>
          아래 버튼을 누르면 퀴즈 풀기를 시작합니다.
        </p>
        <button
          class="start-button css-1y73gpk"
        >
          퀴즈 풀기
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1itlo10"
    >
      <h1>
        Random Quiz!
      </h1>
      <p>
        아래 버튼을 누르면 퀴즈 풀기를 시작합니다.
      </p>
      <button
        class="start-button css-1y73gpk"
      >
        퀴즈 풀기
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
